src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -585,26 +590,26 @@ public DisposableReturnedArray(int length)
 
             public async ValueTask WriteKeyValueAsync(string key, BlittableJsonReaderObject value)
             {
-                if(_operationContext.readTrx.Disposed)
-                    _operationContext.readTrx = _operationContext.ctx.OpenReadTransaction();
-
-                _compareExchangeAddOrUpdateCommands.Add(new AddOrUpdateCompareExchangeCommand(_database.Name, key, value, 0, _context, RaftIdGenerator.DontCareId, fromBackup: true));
                 if (ClusterTransactionCommand.IsAtomicGuardKey(key, out var docId))
                 {
+                    if (_operationContext.readTrx == null || _operationContext.readTrx.Disposed)
+                        _operationContext.readTrx = _operationContext.ctx.OpenReadTransaction();
+
```

Comment:
Need to drop the using below

Reply:
What do you mean?
In line 652?

Reply:
What do you mean?
We

Reply:
No, line 598 - but you already did that in a later commir